src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -333,34 +337,34 @@ public virtual void GetPolicies()
         public bool IsPropertyPolicy { get; protected set; }
 
         // The name from a Json value. This is cached for performance on first deserialize.
-        public byte[] JsonPropertyName { get; set; }
+        public byte[]? JsonPropertyName { get; set; }
 
         // The name of the property with any casing policy or the name specified from JsonPropertyNameAttribute.
-        public byte[] Name { get; private set; }
-        public string NameAsString { get; private set; }
+        public byte[]? Name { get; private set; }
+        public string? NameAsString { get; private set; }
```

Comment:
I don't think `NameAsString` and `Name` can ever be null.

Reply:
OK, `NameAsString` and `Name` can be null for some cases (where `IsPolicyProperty` is true - for instance root collections, primitive value types, etc.).

But, I believe all the current references/calls to these properties can be either done via `Debug.Assert` or `!`.